### PR TITLE
Fixes log handlers breakage since 18.1

### LINKED ISSUE #324: Erlang 18.1 breaks RabbitMQ logging
When upgrading to 3.5.5 I also upgraded erlang to 0:18.1 installed from esl. Downgrading from 0:18.1 to 0:17.5.3 resolved the issue.

All attempts to log end up in ``startup_log`` getting flooded with exceptions trying to log:

```erlang
Error in log handler
====================
Event: {error,<0.119.0>,
              {<0.3140.0>,
               "Channel error on connection ~p (~s, vhost: '~s', user: '~s'), channel ~p:~n~p~n",
               [<0.3132.0>,<<"10.20.16.209:42447 -> 10.20.16.161:5672">>,
                <<"/">>,<<"guest">>,1,
                {amqp_error,not_found,
                            "home node 'rabbit@rabbit1' of durable queue 'foo.bar' in vhost '/' is down or inaccessible",
                            'queue.declare'}]}}
Error: function_clause
Stack trace: [{error_logger_file_h,write_event,
                  [{<0.45.0>,"/var/log/rabbitmq/rabbit@rabbit3.log",[]},
                   {error,<0.119.0>,
                       {<0.3140.0>,
                        "Channel error on connection ~p (~s, vhost: '~s', user: '~s'), channel ~p:~n~p~n",
                        [<0.3132.0>,
                         <<"10.20.16.209:42447 -> 10.20.16.161:5672">>,
                         <<"/">>,<<"guest">>,1,
                         {amqp_error,not_found,
                             "home node 'rabbit@rabbit1' of durable queue 'foo.bar in vhost '/' is down or inaccessible",
                             'queue.declare'}]}}],
                  [{file,"error_logger_file_h.erl"},{line,114}]},
              {error_logger_file_h,handle_event,2,
                  [{file,"error_logger_file_h.erl"},{line,79}]},
              {rabbit_error_logger_file_h,safe_handle_event,3,[]},
              {gen_event,server_update,4,[{file,"gen_event.erl"},{line,538}]},
              {gen_event,server_notify,4,[{file,"gen_event.erl"},{line,520}]},
              {gen_event,server_notify,4,[{file,"gen_event.erl"},{line,522}]},
              {gen_event,handle_msg,5,[{file,"gen_event.erl"},{line,261}]},
              {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]

Error in log handler
====================
Event: {info_msg,<0.119.0>,
                 {<0.3132.0>,"closing AMQP connection ~p (~s)~n",
                  [<0.3132.0>,"10.20.16.209:42447 -> 10.20.16.161:5672"]}}
Error: function_clause
Stack trace: [{error_logger_file_h,write_event,
                  [{<0.45.0>,"/var/log/rabbitmq/rabbit@rabbit3.log",[]},
                   {info_msg,<0.119.0>,
                       {<0.3132.0>,"closing AMQP connection ~p (~s)~n",
                        [<0.3132.0>,
                         "10.20.16.209:42447 -> 10.20.16.161:5672"]}}],
                  [{file,"error_logger_file_h.erl"},{line,114}]},
              {error_logger_file_h,handle_event,2,
                  [{file,"error_logger_file_h.erl"},{line,79}]},
              {rabbit_error_logger_file_h,safe_handle_event,3,[]},
              {gen_event,server_update,4,[{file,"gen_event.erl"},{line,538}]},
              {gen_event,server_notify,4,[{file,"gen_event.erl"},{line,520}]},
              {gen_event,server_notify,4,[{file,"gen_event.erl"},{line,522}]},
              {gen_event,handle_msg,5,[{file,"gen_event.erl"},{line,261}]},
              {proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]
```

### PR DESCRIPTION
**Pull request by videlalvaro**
Fixes #324

### CODE REVIEW

**Comment by videlalvaro** [comment]
Note that until this PR is merged, RabbitMQ's Erlang unit tests will fail

**Comment by michaelklishin** [comment]
This doesn't work on version `18.0.3`, which is an invalid argument for `erlang:list_to_float/1`.

**Comment by videlalvaro** [comment]
@michaelklishin updated the PR

File: src/rabbit_error_logger_file_h.erl [hunk 1]
```diff
@@ -65,18 +80,51 @@ init(File) ->
...
+                    _    ->
+                        {Fd, File, PrevHandler}
+                end,
+            {ok, State};
+        Error    -> Error
+    end.
+
+%% OTP 18.1 introduced the new #st record.
+%% TODO we should use a proper Semver library.
+otp_release_181_or_newer() ->
```
Comment (videlalvaro) [inline src/rabbit_error_logger_file_h.erl]: I feel I need to add this here

File: src/rabbit_error_logger_file_h.erl [hunk 2]
```diff
@@ -65,18 +80,51 @@ init(File) ->
...
+    try
+        case string:tokens(rabbit_misc:otp_release(), ".") of
+            [Maj, Min | _ ] ->
+                Maj1 = list_to_integer(Maj),
+                Min1 = list_to_integer(Min),
+                Maj1 >= 18 andalso Min1 >= 1;
```
Comment (videlalvaro) [inline src/rabbit_error_logger_file_h.erl]: This is clearly wrong, since it will report 19.0 as being lower than 18.1

Comment (michaelklishin) [inline src/rabbit_error_logger_file_h.erl]: We need to account for `R13…` to `R16…` before splitting, too.

**Comment by videlalvaro** [comment]
ready for another round @michaelklishin

**Pull request merged**